internal/app/handler/handler.go

```diff
@@ -43,10 +55,186 @@ func CreateShortLink(w http.ResponseWriter, r *http.Request) {
+		shortLink = helper.Concat2builder(s.Config.GetConfig().ServerAddress, "/", url.Code)
+		w.WriteHeader(http.StatusConflict)
+	} else {
+
+		w.WriteHeader(http.StatusCreated)
+	}
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
 
-	w.WriteHeader(http.StatusCreated)
-	shortLink := helper.Concat2builder("http://", config.Domain, "/", code)
 	w.Write([]byte(shortLink))
 }
+
+func (s *Server) ShortHandler(w http.ResponseWriter, r *http.Request) {
+	user := r.Context().Value(middleware.Key).([]byte)
+
+	reader, err := helper.ReadRequest(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	code := helper.NewCode()
+
+	var ret [16]byte
+	copy(ret[:], user)
+
+	var newURL = model.URLLink{ID: time.Now().UnixNano(), Code: code, UserID: ret}
+
+	err = json.NewDecoder(reader).Decode(&newURL)
+	defer r.Body.Close()
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	u := helper.IsURL(newURL.Long)
+
+	if !u {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	shortLink := helper.Concat2builder(s.Config.GetConfig().ServerAddress, "/", code)
+	newURL.ShortLink = shortLink
+
+	codeEvents := ""
+	codeEvents, err = s.Service.AddLink(newURL)
+	var ReturnURL = model.URLLink{}
+	w.Header().Set("Content-Type", "application/json")
+	if codeEvents == "23505" {
+		url, _ := s.Service.GetLinkByLong(newURL.Long)
+
+		shortLink = helper.Concat2builder(s.Config.GetConfig().ServerAddress, "/", url.Code)
+		ReturnURL.ShortLink = shortLink
+		ReturnURL.UserID = url.UserID
+		w.WriteHeader(http.StatusConflict)
+	} else {
+		ReturnURL.ShortLink = newURL.ShortLink
+		ReturnURL.UserID = newURL.UserID
+		w.WriteHeader(http.StatusCreated)
+	}
+
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	res, err := json.Marshal(ReturnURL)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	w.Write(res)
+}
+
+func (s *Server) GetUserURL(w http.ResponseWriter, r *http.Request) {
+	user := r.Context().Value(middleware.Key).([]byte)
+	var ret [16]byte
+	copy(ret[:], user)
+	urls, err := s.Service.GetLinksUser(ret)
+
+	if err != nil {
+		w.WriteHeader(http.StatusNoContent)
+		return
+	}
+
+	res, err := json.Marshal(urls)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(res)
+}
+
+func (s *Server) BatchShorten(w http.ResponseWriter, r *http.Request) {
+	user := r.Context().Value(middleware.Key).([]byte)
+	reader, err := helper.ReadRequest(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	var ret [16]byte
+	copy(ret[:], user)
+
+	type UserURLLinks struct {
+		Correlation string `json:"correlation_id,omitempty"`
+		OriginalURL string `json:"original_url,omitempty"`
+	}
+
+	type UserURLLinksResponse struct {
+		Correlation string `json:"correlation_id,omitempty"`
+		ShortURL    string `json:"short_url,omitempty"`
+	}
+
+	var arrURL []UserURLLinks
```

Comment:
```suggestion
	var arrURL []*UserURLLinks
```
Иначе при том range, который используется дальше, будут создаваться копии каждой структуры.